test/trafficsplit/trafficsplit_test.go

```diff
@@ -0,0 +1,247 @@
+package trafficsplit
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/linkerd/linkerd2/testutil"
+)
+
+var TestHelper *testutil.TestHelper
+
+func TestMain(m *testing.M) {
+	TestHelper = testutil.NewTestHelper()
+	os.Exit(m.Run())
+}
+
+type statTsRow struct {
+	name       string
+	apex       string
+	leaf       string
+	weight     string
+	success    string
+	rps        string
+	latencyP50 string
+	latencyP95 string
+	latencyP99 string
+}
+
+func parseStatTsRow(out string, expectedRowCount, expectedColumnCount int) (map[string]*statTsRow, error) {
+	rows, err := testutil.CheckRowCount(out, expectedRowCount)
+	if err != nil {
+		return nil, err
+	}
+
+	statRows := make(map[string]*statTsRow)
+
+	for _, row := range rows {
+		fields := strings.Fields(row)
+
+		if len(fields) != expectedColumnCount {
+			return nil, fmt.Errorf(
+				"Expected [%d] columns in stat output, got [%d]; full output:\n%s",
+				expectedColumnCount, len(fields), row)
+		}
+
+		row := &statTsRow{
+			name:       fields[0],
+			apex:       fields[1],
+			leaf:       fields[2],
+			weight:     fields[3],
+			success:    fields[4],
+			rps:        fields[5],
+			latencyP50: fields[6],
+			latencyP95: fields[7],
+			latencyP99: fields[8],
+		}
+
+		statRows[row.leaf] = row
+
+	}
+	return statRows, nil
+}
+
+func statTrafficSplit(from string, ns string) (map[string]*statTsRow, error) {
+	cmd := []string{"stat", "ts", "--from", from, "--namespace", ns}
+	stdOut, _, err := TestHelper.LinkerdRun(cmd...)
+	if err != nil {
+		return nil, err
+	}
+	return parseStatTsRow(stdOut, 2, 9)
+}
+
+func validateTrafficSplit(actual *statTsRow, expected *statTsRow) error {
+
+	if actual.name != expected.name && expected.name != "" {
+		return fmt.Errorf("expected name '%s' for leaf %s, got '%s'", expected.name, expected.leaf, actual.name)
+	}
+
+	if actual.apex != expected.apex && expected.apex != "" {
+		return fmt.Errorf("expected apex '%s' for leaf %s, got '%s'", expected.apex, expected.leaf, actual.apex)
+	}
+
+	if actual.weight != expected.weight && expected.weight != "" {
+		return fmt.Errorf("expected weight '%s' for leaf %s, got '%s'", expected.weight, expected.leaf, actual.weight)
+	}
+
+	if actual.success != expected.success && expected.success != "" {
+		return fmt.Errorf("expected success '%s' for leaf %s, got '%s'", expected.success, expected.leaf, actual.success)
+	}
+
+	if actual.rps != expected.rps && expected.rps != "" {
```

Comment:
rps will never match exactly, but we can make sure that if expected rps is 0 then actual rps must be 0 and if expected rps is non-zero then actual rps must be non-zero.

Reply:
I agree with @adleong that it would be helpful to have some checks around RPS being non-zero. 

Also, although we can't match RPS, we are able to calculate the relative percentages of RPS expected, from the weights. Could we compare the RPS of each leaf? If there is a `500m/500m` split we could divide `leaf1` by `leaf2` and check to see that the result was between `9.9` and `1.1`. Would this be practical, or too vulnerable to flakiness?

If the above idea is practical, we could test additional traffic ratios, such as `200m/800m`.

The reason I'm thinking about this is because of the conversation in #3562 - what additional checks can we add to this test to reflect an actual traffic split scenario?

Reply:
@scottcarol I think although this is a good idea, it probably is trying to be a bit too clever and might introduce flakiness. The purpose of the test as its written is to ensure that there is traffic flowing to both services. But since one of them is an always failing one, it might very well be the case that its RPS  might be a bit different (just a guess). But to illustrate this with an example. After I modify the split resource weights to be equal, this is what I get as an output at some point: 

NAME                    APEX          LEAF          WEIGHT   SUCCESS      RPS   LATENCY_P50   LATENCY_P95   LATENCY_P99
```
backend-traffic-split   backend-svc   backend-svc     500m   100.00%   0.4rps           8ms          17ms          20ms
backend-traffic-split   backend-svc   failing-svc     500m     0.00%   0.2rps           6ms          27ms          30ms
```

Reply:
Yes, it's probably not practical to compare theoretical/real RPS. If it was a priority in the future to measure RPS changes, we *could* calculate the general trend in RPS over time and report whether each leaf's trend was positive or negative when switching from `200m/300m` to `500m/0m`, for example. This would be slightly more specific than measuring for zeroRPS. Completely unnecessary for this test and PR though!